src/main/java/org/threadly/litesockets/TCPClient.java

```diff
@@ -206,16 +206,21 @@ protected void doSocketWrite(boolean doLocal) {
 
   @Override
   public ListenableFuture<?> write(final ByteBuffer bb) {
+    return write(new MergedByteBuffers(false, bb));
```

Comment:
Thoughts on how we could avoid wrapping this??  This is not the lightest weight wrapper class, in part because of the `ArrayDeque` inside (we could consider making this constructor use a `LinkedList` or something).

It would be nice if the `ByteBuffer` write did not incur additional overhead, but I am not sure if it's worth the effort either.